src/Fixer/PhpTag/ShortEchoTagFixer.php

```diff
@@ -0,0 +1,205 @@
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use SplFileInfo;
+
+/**
+ * @author Michele Locati <michele@locati.it>
+ */
+final class ShortEchoTagFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    /** @internal */
+    const OPTION_FORMAT = 'format';
+
+    /** @internal */
+    const OPTION_SHORT_ALWAYS = 'short-always';
+
+    /** @internal */
+    const OPTION_LONG_FUNCTION = 'long-function';
+
+    /** @internal */
+    const FORMAT_SHORT = 'short';
+
+    /** @internal */
+    const FORMAT_LONG = 'long';
+
+    /** @internal */
+    const LONG_FUNCTION_ECHO = 'echo';
+
+    /** @internal */
+    const LONG_FUNCTION_PRINT = 'print';
+
+    private $supportedFormatOptions = [
+        self::FORMAT_SHORT,
+        self::FORMAT_LONG,
+    ];
+
+    private $supportedLongFunctionOptions = [
+        self::LONG_FUNCTION_ECHO,
+        self::LONG_FUNCTION_PRINT,
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        $sample = <<<'EOT'
+<?=1?>
+<?php print '2' . '3'; ?>
+<?php print '2' . '3'; someFunction(); ?>
+
+EOT
+        ;
+
+        return new FixerDefinition(
+            'Replace short-echo `<?=` with long format `<?php echo`/`<?php print` syntax, or vice-versa.',
+            [
+                new CodeSample($sample),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_LONG]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_LONG, self::OPTION_LONG_FUNCTION => self::LONG_FUNCTION_PRINT]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_SHORT]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_SHORT, self::OPTION_SHORT_ALWAYS => true]),
+            ],
+            null
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        if (self::FORMAT_SHORT === $this->configuration[self::OPTION_FORMAT]) {
+            if (null !== $tokens->findSequence([[\T_OPEN_TAG], [\T_ECHO]])) {
+                return true;
+            }
+            if (null !== $tokens->findSequence([[\T_OPEN_TAG], [\T_PRINT]])) {
+                return true;
+            }
+
+            return false;
+        }
+
+        return $tokens->isTokenKindFound(\T_OPEN_TAG_WITH_ECHO);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder(self::OPTION_FORMAT, 'How the fixer should process short/long echo tags'))
+                ->setAllowedValues($this->supportedFormatOptions)
+                ->setDefault(self::FORMAT_SHORT)
```

Comment:
Why not `self::FORMAT_SHORT`?

Reply:
> Why not `self::FORMAT_SHORT`?

@kubawerlos I'm not sure I understand... it *is* `self::FORMAT_SHORT`...

Reply:
@mlocati yes, sorry, I meant why not `self::FORMAT_LONG` - it would be consistent with `no_short_echo_tag`

Reply:
@kubawerlos I agree: see 451129f30ac68adfa68723b156872b18c5293b14